p2p/node_test.go

```diff
@@ -87,43 +87,117 @@ func newTestNode(t *testing.T) *Node {
 		RendezvousString: testRendezvousString,
 		UseBootstrapList: false,
 	}
-	node, err := New(config)
+	node, err := New(ctx, config)
 	require.NoError(t, err)
-	return node
-}
-
-func createTwoConnectedTestNodes(t *testing.T, notifee p2pnet.Notifiee) (*Node, *Node) {
-	t.Helper()
-	// Create two nodes and add each one to the other's peerstore and connect
-	// them.
-	node0 := newTestNode(t)
-	node1 := newTestNode(t)
 
 	// If notifee is not nil, set up *both* hosts to use it.
 	if notifee != nil {
-		node0.host.Network().Notify(notifee)
-		node1.host.Network().Notify(notifee)
+		node.host.Network().Notify(notifee)
 	}
-	connectCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
+
+	return node
+}
+
+func connectTestNodes(t *testing.T, node0, node1 *Node) {
```

Comment:
I had to change the tests around quite a bit because we need more fine-grained control around when and how `New`, `Start`, and `Connect` are called.